src/reconcile/rec_write.c

```diff
@@ -1835,6 +1835,13 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
     verify_image = true;
 #endif
 
+    /*
+     * If reconciliation requires multiple blocks and checkpoint is running we'll eventually fail.
+     * It's possible this is a big page that will take a lot of writes, avoid wasted work.
+     */
+    if (!last_block && __wt_btree_ckpt_blocks_eviction(session))
```

Comment:
Looks like this introduces an issue on shutdown that isn't OK - so it probably needs a tweak where __wt_btree_ckpt_blocks_eviction might need to know when it's being called?

Reply:
Checking `WT_CONN_CLOSING_TIMESTAMP` in `__wt_btree_can_evict_dirty()` seems wrong to me, we call that function in the split code, and it shouldn't be checking the connection close flags.

The change I pushed in efdff20 ignores that flag, but should still catch the big ticket item, which is large pages.

FWIW, we could actually tighten down all of the reconciliation and page split checks because we don't care if checkpoint has started, we care if checkpoint has moved into the `WT_BTREE_SYNC_RUNNING` phase. That might reduce the competition for blocks in the tree during checkpoint, but almost certainly not enough to matter in any realistic workload.